fapolicy_analyzer/ui/subject_list.py

```diff
@@ -141,6 +155,25 @@ def find_db_trust(subject):
         if changeset:
             dispatch(apply_changesets(changeset))
 
+    def __show_confirmation_dialog(self, subjects):
+        changeset = None
+        parent = self.get_ref().get_toplevel()
+        confirmationDialog = ConfirmChangeDialog(parent=parent).get_ref()
+        confirm = confirmationDialog.run()
+        confirmationDialog.destroy()
+
+        trust = self.find_db_trust(subjects[0])
+        if confirm == 1:
+            changeset = Changeset()
+            paths = [subject.file for subject in subjects]
+            if isinstance(trust, Trust):
+                changeset.del_trust(*paths)
+            else:
+                changeset.add_trust(*paths)
```

Comment:
I still don't think you logic is quite right here. Not that you have the trust you need to look at the trust.status to determine the action.

trust is None = Add
trust is t and subject.trust is st = no action (there is nothing we can change about a file trusted in the system db)
trust is t and subject.trust is at = Del
trust is not t = Add
You can look at TrustReconciliationDialog.__load_details for how I'm setting the button visibility for this.